test/common/utxobased/db/NewProcessor.spec.ts

```diff
@@ -0,0 +1,322 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import * as chai from 'chai'
+import { expect } from 'chai'
+import { makeMemoryDisklet } from 'disklet'
+
+import {
+  makeNewProcessor,
+  NewProcessor
+} from '../../../../src/common/utxobased/db/newProcessor'
+import {
+  IProcessorTransaction,
+  ITransactionInput,
+  ITransactionOutput
+} from '../../../../src/common/utxobased/db/types'
+
+chai.should()
+
+describe('Processor transactions tests', () => {
+  let processor: NewProcessor
+
+  function assertNumTransactions(expectedNum: number): void {
+    const num = processor.numTransactions()
+    expect(num).eql(expectedNum)
+  }
+
+  it('empty transaction baselets', async () => {
+    const storage = {}
+    const disklet = makeMemoryDisklet(storage)
+
+    processor = await makeNewProcessor({ disklet })
+    assertNumTransactions(0)
+  })
+
+  it('insert a transaction to transaction baselets', async () => {
+    const storage = {}
+    const disklet = makeMemoryDisklet(storage)
+    processor = await makeNewProcessor({ disklet })
+
+    const input1: ITransactionInput = {
+      txId: 'random',
+      outputIndex: 0,
+      scriptPubkey: 'pubkeyin1',
+      n: 0,
+      amount: '1'
+    }
+    const output1: ITransactionOutput = {
+      amount: '1',
+      n: 0,
+      scriptPubkey: 'pubkeyout1'
+    }
+    const output2: ITransactionOutput = {
+      amount: '1',
+      n: 1,
+      scriptPubkey: 'pubkeyout2'
+    }
+    const transaction1: IProcessorTransaction = {
+      txid: 'transaction1',
+      hex: '',
+      blockHeight: 1,
+      date: new Date(10).getTime(),
+      fees: '1',
+      inputs: [input1],
+      outputs: [output1, output2],
+      ourIns: [],
+      ourOuts: [],
+      ourAmount: '0'
+    }
+
+    await processor.insertTransaction({
+      tx: transaction1,
+      scriptPubkey: 'pubkeyout1'
```

Comment:
Should use `output1.scriptPubkey` instead of duplicating the string? The same goes for other strings below; the reason for this is to make sure that if a string ever changes in the spec, it wont break and require many changes; it also voids the possibility of typos.

Reply:
It does convey that they may not be related, and that we are passing in the matching script pubkey 'by chance' . But at the same time, I agree with you, since we are testing this very case (e.g. scriptPubkey matches with the one in output 1). Changed it to match your suggestion and included references to the txid's as well where applicable.